src/libcrun/criu.c

```diff
@@ -673,6 +681,30 @@ libcrun_container_restore_linux_criu (libcrun_container_status_t *status, libcru
 
           criu_add_inherit_fd (inherit_new_pid_fd, CRIU_EXT_PIDNS);
         }
+
+      if (value == CLONE_NEWTIME && def->linux->namespaces[i]->path != NULL)
+        {
+          if (join_ns_support)
+            criu_join_ns_add("time", def->linux->namespaces[i]->path, NULL);
```

Comment:
do we need a `configure` time check for this function?

Reply:
I've added a check for the minimum version of criu. The function itself just sends the strings `"time"` and `"/proc/<pid>/ns/time"` over RPC to criu. Then criu calls `setns()` to restore the process tree in the correct namespace.

I think we don't need a `configure` check for `CLONE_NEWTIME` because the compilation should work even if timens is not supported. At run-time, restore should fail if the container specifies shared timens and the kernel doesn't support this.